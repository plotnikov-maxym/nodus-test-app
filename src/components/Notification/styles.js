export const styles = ({spacing: {unit}}) => ({
  notification: {
    position: "absolute",
    left: 0,
    top: 0,
    zIndex: 100,
    backgroundColor: "transparent",
    opacity: 0.9,
    minHeight: unit * 4,
    width: "100%",
    display: "flex",
    flexDirection: "row",
    justifyContent: "space-between",
    alignItems: "center",
    padding: `0 ${unit * 2}px`,
  },
  error: {
    backgroundColor: "#f44336",
  },
  info: {backgroundColor: "#4caf50"},
  text: {
    padding: 0,
    color: "white",
    flexGrow: 3,
    margin: 0,
  },
});
