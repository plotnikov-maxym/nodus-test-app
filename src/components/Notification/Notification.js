import React from "react";
import {oneOf, shape, string} from "prop-types";
import classNames from "classnames";
import {NOTIFICATION_TYPES} from "../../constants/notifications";
import {styles} from "./styles";
import {withStyles} from "@material-ui/core/styles";
import Typography from "@material-ui/core/Typography";

export const Notification = ({classes, notification}) => {
  if (notification) {
    const {type, message} = notification;
    const notificationClass = classes[type];

    if (message) {
      return (
        <div className={classNames(classes.container, notificationClass)}>
          {message && (
            <Typography variant="body1" className={classes.text}>
              {message}
            </Typography>
          )}
        </div>
      );
    }
  }

  return null;
};

Notification.propTypes = {
  notification: shape({
    type: oneOf(NOTIFICATION_TYPES),
    message: string,
  }),
};

export default withStyles(styles)(Notification);
