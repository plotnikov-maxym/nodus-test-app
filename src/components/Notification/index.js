import {connect} from "react-redux";
import Notification from "./Notification";

const mapStateToProps = state => ({
  notification: state.notifications,
});
export default connect(mapStateToProps)(Notification);
