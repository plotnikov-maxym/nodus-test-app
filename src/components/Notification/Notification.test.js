import React from "react";
import renderer from "react-test-renderer";
import {extend} from "underscore";
import {Notification} from "./Notification";
import {createShallow} from "@material-ui/core/test-utils";
import Typography from "@material-ui/core/Typography";

describe("<Notification />", () => {
  let shallow;

  beforeAll(() => {
    shallow = createShallow();
  });

  describe("Notification component", () => {
    it("matches snapshot", async () => {
      const wrapper = await renderComponent();
      expect(wrapper).toMatchSnapshot();
    });

    it("renders properly the correct color on error", async () => {
      const notification = {
        type: "error",
        message: "Test Message",
      };
      const wrapper = await getComponent({notification});
      expect(wrapper.find(Typography)).toExist();
      expect(wrapper.find(Typography).props().children).toEqual(
        notification.message,
      );
      const {classes} = getMockProps();
      expect(wrapper.find("div").prop("className")).toEqual(
        `${classes.notification} ${classes.error}`,
      );
    });

    it("does not renders if notification prop is absent", async () => {
      const notification = null;
      const wrapper = await getComponent({notification});
      expect(wrapper.find(Typography)).not.toExist();
    });
  });

  const getComponent = props => {
    const parsedProps = extend(getMockProps(), props);

    return shallow(<Notification {...parsedProps} />);
  };

  const renderComponent = props => {
    const parsedProps = extend(getMockProps(), props);

    return renderer.create(<Notification {...parsedProps} />).toJSON();
  };

  const getMockProps = () => ({
    classes: {
      notification: "notification-121",
      error: "error-122",
      info: "info-124",
      text: "text-123",
    },
    notification: {
      type: "info",
      message: "Test Message",
    },
  });
});
