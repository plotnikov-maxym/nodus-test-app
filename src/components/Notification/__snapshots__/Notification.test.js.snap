// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Notification /> Notification component matches snapshot 1`] = `
<div
  className="info-124"
>
  <p
    className="MuiTypography-root-1 MuiTypography-body1-10 text-123"
  >
    Test Message
  </p>
</div>
`;
